src/bp/ui-shared-lite/Payloads/index.ts

```diff
@@ -0,0 +1,162 @@
+    }
+  }
+  return {
+    type: 'custom',
+    module: 'channel-web',
+    component: 'QuickReplies',
+    quick_replies: content.choices.map(c => ({
+      title: c.title,
+      payload: c.value.toUpperCase()
+    })),
+    disableFreeText: (content as any).disableFreeText,
+    wrapped: {
+      type: 'text',
+      ...omit(content, 'choices', 'type')
+    }
+  }
+}
+
+const renderDropdownPayload = (content: any) => {
+  // TODO: add typings for dropdowns
+  return {
+    type: 'custom',
+    module: 'extensions',
+    component: 'Dropdown',
+    message: content.message,
+    buttonText: content.buttonText,
+    displayInKeyboard: content.displayInKeyboard,
+    options: content.options,
+    allowCreation: content.allowCreation,
+    allowMultiple: content.allowMultiple,
+    width: content.width,
+    collectFeedback: content.collectFeedback,
+    placeholderText: content.placeholderText,
+    markdown: content.markdown
+  }
+}
+
+const renderImagePayload = (content: sdk.ImageContent) => {
+  return {
+    type: 'file',
+    title: content.title,
+    url: formatUrl('', content.image),
+    collectFeedback: (content as any).collectFeedback
+  }
+}
+
+const renderAudioPayload = (content: sdk.AudioContent) => {
+  return {
+    type: 'audio',
+    title: content.title,
+    url: formatUrl('', content.audio),
+    collectFeedback: (content as any).collectFeedback
+  }
+}
+
+const renderVideoPayload = (content: sdk.VideoContent) => {
+  return {
+    type: 'video',
+    title: content.title,
+    url: formatUrl('', content.video),
+    collectFeedback: (content as any).collectFeedback
+  }
+}
+
+const renderFilePayload = (content: any) => {
+  return {
+    type: 'video',
+    title: content.title,
+    url: formatUrl('', content.file),
+    collectFeedback: content.collectFeedback
+  }
+}
+
+const renderCarouselPayload = (content: sdk.CarouselContent) => {
+  return {
+    text: ' ',
+    type: 'carousel',
+    collectFeedback: (content as any).collectFeedback,
+    elements: content.items.map(card => ({
+      title: card.title,
+      picture: card.image ? formatUrl('', card.image) : null,
+      subtitle: card.subtitle,
+      buttons: (card.actions || []).map(a => {
+        if (a.action === 'Say something') {
+          return {
+            type: 'say_something',
+            title: a.title,
+            text: (a as sdk.ActionSaySomething).text
+          }
+        } else if (a.action === 'Open URL') {
+          return {
+            type: 'open_url',
+            title: a.title,
+            url: (a as sdk.ActionOpenURL)?.url.replace('BOT_URL', '')
+          }
+        } else if (a.action === 'Postback') {
+          return {
+            type: 'postback',
+            title: a.title,
+            payload: (a as sdk.ActionPostback).payload
+          }
+        } else {
+          throw new Error(`Webchat carousel does not support "${a.action}" action-buttons at the moment`)
+        }
+      })
+    }))
+  }
+}
```

Comment:
Okay.. So basically the render logic was just removed from the content types to be moved on ui-shared.. ? Why not simply put this in channel-web ? We should just directly use the payloads saved by the studio instead of having to "re-render" it

Reply:
It was done this way to keep compatibility with old conversations (that are still stored with the old payloads). It would also probably broken lots of things to change every component in the channel-web. This solution is pretty simple and works for other modules that render messages like hitl and bot-improvement.